oligo_designer_toolsuite/oligo_specificity_filter/_filter_blastn.py

```diff
@@ -0,0 +1,289 @@
+
+        self.dir_probes = os.path.join(self.dir_output, "probes_blast")
+        Path(self.dir_probes).mkdir(parents=True, exist_ok=True)
+
+        self.file_removed_genes = os.path.join(
+            self.dir_probes, "genes_with_insufficient_probes.txt"
+        )
+
+    def run_blast_search(self):
+        """Run BlastN alignment tool to find regions of local similarity between sequences, where sequences are probes and transcripts.
+        BlastN identifies the transcript regions where probes match with a certain coverage and similarity.
+        """
+
+        def _run_blast(batch_id):
+            """Run BlastN alignment search for all probes of one batch
+            :param batch_id: Batch ID.
+            :type batch_id: int
+            """
+            for subbatch_id in range(self.number_subbatches):
+                file_probe_fasta_batch = os.path.join(
+                    self.dir_annotations,
+                    "probes_sequence_batch{}_{}.fna".format(batch_id, subbatch_id),
+                )
+                file_blast_batch = os.path.join(
+                    self.dir_blast, "blast_batch{}_{}.txt".format(batch_id, subbatch_id)
+                )
+
+                cmd = NcbiblastnCommandline(
+                    query=file_probe_fasta_batch,
+                    db=self.file_transcriptome_fasta,
+                    outfmt="10 qseqid sseqid length qstart qend qlen",
+                    out=file_blast_batch,
+                    strand="plus",
+                    word_size=self.word_size,
+                    perc_identity=self.percent_identity,
+                    num_threads=self.n_jobs,
+                )
+                out, err = cmd()
+
+        # create blast database
+        self.logging.info("Creating blast database")
+        start_time = time.perf_counter()
+        cmd = NcbimakeblastdbCommandline(
+            input_file=self.file_transcriptome_fasta, dbtype="nucl"
+        )
+        out, err = cmd()
+
+        finish_time = time.perf_counter()
+
+        self.logging.info(f"Blast database created in {finish_time-start_time} seconds")
+
+        # run blast in parallel
+        start_time = time.perf_counter()
+        with parallel_backend("loky"):
+            Parallel()(delayed(_run_blast)(batch_id) for batch_id in range(self.n_jobs))
+
+        finish_time = time.perf_counter()
+        self.logging.info(
+            f"Blast alignmnet search finished in {finish_time-start_time} seconds"
+        )
+
+    def filter_probes_by_blast_results(self):
+        """Process the output of the BlastN alignment search"""
+
+        def _process_blast_results(batch_id):
+
+            probes_info = _load_probes_info(self, batch_id)
+
+            num_probes_wo_match = 0
+            for subbatch_id in range(self.number_subbatches):
+                blast_results = _read_blast_output(batch_id, subbatch_id)
+                num_probes_wo_match += _filter_probes_blast(probes_info, blast_results)
```

Comment:
we should not need that anymore when we simply delete the probes from the probe database, like in the other filters